### DISCUSSION TITLE
run the backend and frontend separately

### DISCUSSION THREAD

**Discussion by volodgo** [eclipse-sirius/sirius-components#3608] [Q&A] [answered]
Hello!
I want to run the backend and frontend separately.
I can do this for the front "turbo run start" and it works.
How can I do this separately for the backend, so that I can develop the frontend separately from the backend.

**Comment by pcdavid** [comment]
It depends on which IDE you use to develop on the backend, but in general there is nothing special to do.
Simply start the `sirius-web-sample-application` Spring Boot application, for example using the Eclipse-based Spring Tools Suite:

![screenshot-2024-06-12_09-02](https://github.com/eclipse-sirius/sirius-web/assets/10608/b686f576-dad8-49ff-a844-ba207c2c0c75)

Of course you must also have a PostgreSQL database running at [the location expected in the app's configuration properties](https://github.com/eclipse-sirius/sirius-web/blob/master/packages/sirius-web/backend/sirius-web-sample-application/src/main/resources/application-dev.properties#L38), see [the README](https://github.com/eclipse-sirius/sirius-web?tab=readme-ov-file#quick-start) for an example Docker command-line.

**Comment by volodgo** [comment, marked as answer]
OK, thanks for answer!
I don't want to change the backend, I take it as is. I found the following solution for myself
I commented out the sirius-web-frontend dependency so it doesn't run the frontend
![image](https://github.com/eclipse-sirius/sirius-web/assets/24474478/6715f5aa-d83d-4c18-b47a-a11d77b02153)
I don't understand Java. Did I turn off the front module correctly?
Next, I modified sirius-web/packages/sirius-web/frontend/sirius-web/vite.config.js so that requests are proxied to a different url
![image](https://github.com/eclipse-sirius/sirius-web/assets/24474478/f8741887-97d3-40c2-a4e6-e3ee1f3ac45a)
And changed sirius-web/packages/sirius-web/frontend/sirius-web/.env.development
![image](https://github.com/eclipse-sirius/sirius-web/assets/24474478/9332b8e7-8437-47fb-a39a-80a57caebb99)

**Comment by pcdavid** [comment]
> I don't understand Java. Did I turn off the front module correctly?

Yes, but it should not be necessary. This is just to provide the built version of the frontend (HTML/CSS/JS/images) as static assets for the backend to serve at :8080 when in production.

In your case, if you run your front on e.g. :5173 and use that from your browser, it will talk to the backend APIs at :8080 but never ask for the frontend assets there.

**Comment by volodgo** [comment]
Yes, you are right, this is not necessary)
